python/modules/tauMVA.py

```diff
@@ -0,0 +1,137 @@
+from importlib import import_module
+from os import system, environ
+
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection, Object
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+from PhysicsTools.NanoAODTools.postprocessing.tools import deltaPhi, deltaR, closest
+from PhysicsTools.NanoSUSYTools.modules.xgbHelper import XGBHelper
+
+class tauMVA(Module):
+    def __init__(self):
+	self.writeHistFile=True
+	self.metBranchName = "MET"
+	#still trying to find appropriate cut, but the this is the best training model
+	self.tauMVADisc = 0.71
+	self.bdt_file = environ["CMSSW_BASE"] + "/src/PhysicsTools/NanoSUSYTools/data/tauMVA/tauMVA-xgb_nvar13_eta0_003000_maxdepth10.model"
+	self.bdt_vars = ['pt', 'abseta', 'chiso0p1', 'chiso0p2', 'chiso0p3', 'chiso0p4', 'totiso0p1', 'totiso0p2', 'totiso0p3', 'totiso0p4', 'neartrkdr', 'contjetdr', 'contjetcsv']
+	self.xgb = XGBHelper(self.bdt_file, self.bdt_vars)
+
+    def beginJob(self,histFile=None,histDirName=None):
+   	pass
+    def endJob(self):
+	pass
+
+    def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        self.out = wrappedOutputTree
+        self.out.branch("mt", 			"F", lenVar="nPFcand")
+	self.out.branch("taumva", 		"F", lenVar="nPFcand")
+	self.out.branch("TauMVA_Stop0l",        "I")
+
+    def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        pass
+
+    def SelTauMVA(self, mva):
+	if mva > self.tauMVADisc:
+		return True
+	else:
+		return False
+
+    def getNearPhotonIndex(self, pfc, pfcands):
+	minPhotonPt = 0.5
+	maxPhotonDR = 0.2
+	photonInd = -1
+	maxPhotonPT = 0.0
+	
+	for ic in xrange(len(pfcands)):
+		c = pfcands[ic]
+		if(c.nearphopt < minPhotonPt): continue
+		dr = deltaR(c.eta, c.phi, pfc.nearphoeta, pfc.nearphophi)
+		if(dr > maxPhotonDR): continue;
+		if(c.nearphopt > maxPhotonPT):
+			maxPhotonPT = c.pt;
+			photonInd = ic;
+	
+	return photonInd;
+
+    def transverseMass(self, visible, invisible):
+	cosDPhi   = np.cos( deltaPhi(visible.Phi(), invisible.phi) );
+	return np.sqrt( 2 * visible.Pt() * invisible.pt * (1 - cosDPhi) );
+        
+    def computeMT(self, pfc, met, pfcands):
+	photonInd = self.getNearPhotonIndex(pfc, pfcands);
+	candP4 = ROOT.TLorentzVector()
+	candP4.SetPtEtaPhiM(pfc.pt, pfc.eta, pfc.phi, pfc.mass)
+	if(photonInd > -1): 
+		pfcand_buff = ROOT.TLorentzVector()
+		pfcand_buff.SetPtEtaPhiM(pfcands[photonInd].pt, pfcands[photonInd].eta, pfcands[photonInd].phi, pfcands[photonInd].mass)
+		candP4+=pfcand_buff;
+	return self.transverseMass(candP4, met);
+
+    def analyze(self, event):
+        ## Getting objects
+	met	  = Object(event, self.metBranchName)
+	jets	  = Collection(event, "Jet")
+	pfcand    = Collection(event, "PFcand")
+	eventNum  = event.event
+
+        pfchargedhads = []
+	mva = {}
+	mva_ = []
+	mt_ = []
+	for pfc in pfcand:
+		mva_buff = 0.0
+		mt = 0.0
+		if(pfc.pt > 10.0 and abs(pfc.eta) < 2.4 and abs(pfc.dz) < 0.2):
```

Comment:
There is a pt>10 and abs(eta) < 2.4 and abs(dz) < 0.2 cut on the mva calculation so you can add this somehow